src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -746,25 +748,50 @@ GenerateSizeQueryOnMultiplePlacements(List *shardIntervalList,
 		char *shardQualifiedName = quote_qualified_identifier(schemaName, shardName);
 		char *quotedShardName = quote_literal_cstr(shardQualifiedName);
 
+		/* for partitoned tables, call worker_partitioned_... size functions */
 		if (optimizePartitionCalculations && PartitionedTable(shardInterval->relationId))
 		{
 			appendStringInfo(selectQuery, GetWorkerPartitionedSizeUDFNameBySizeQueryType(
 								 sizeQueryType), quotedShardName);
+			appendStringInfo(selectQuery, " + ");
 		}
+		/* for non-partitioned tables, add them into a list and then calculate using a SUM */
 		else
 		{
-			appendStringInfo(selectQuery, GetSizeQueryBySizeQueryType(sizeQueryType),
-							 quotedShardName);
+			nonPartitionedShardNames = lappend(nonPartitionedShardNames, quotedShardName);
 		}
+	}
 
-		appendStringInfo(selectQuery, " + ");
+	if (list_length(nonPartitionedShardNames) < 1)
```

Comment:
nit: I think comparing to 0 is slightly clearer
```suggestion
	if (list_length(nonPartitionedShardNames) == 0)
```